### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by stefonarch** [LBCrion/sfwbar#153]
Yesterday I found sfwbar with 2G of memory after half a day. Now it's already at 240mb in 15 min. I removed the AUR package and compiled to `/usr/local` to make sure this is used.

![memleak-sfwbar](https://github.com/LBCrion/sfwbar/assets/10681413/6157abc7-cd61-4fe7-84c3-aa0b580e7aca)

![screen_area_mer_08:35:36_](https://github.com/LBCrion/sfwbar/assets/10681413/efc0886f-4bf0-4747-bfbb-ea1cdb34112a)

**Comment by LBCrion** [comment]
Let me investigate this. Are you using a default config? If not, would you
mind posting the one you're using?

**Comment by LBCrion** [comment]
Actually, I think I found it. Can you update to the latest git and check if
you still see this?

**Comment by stefonarch** [comment]
```
function("SfwbarInit") {
  SetBarId "bar-0"
  SetLayer "bottom"
  SetExclusiveZone "auto"
  SetMonitor "eDP-1"
  SetBarSize "70%"
}

# Panel layout

layout "sfwbar" {

  taskbar {
    rows = 1
    group = false
    icons = true
    labels = true
    action[2] = Close
    action[3] = Maximize
    title_width= 20
  }

  label { css = "* { -GtkWidget-hexpand: true; min-height: 28px; }" }

}

#CSS
@define-color theme_bg_color #353231;
@define-color theme_fg_color #d1d1d1;
@define-color borders #777777;

window {
  -GtkWidget-direction: bottom;
    background-color: transparent;
}

label {
  font: 18px sans;
  text-shadow: none;
}

button
{
  outline-style: none;
  box-shadow: none;
  background-image: none;
  border: solid 1px transparent;
  border-radius: 6px;
  margin: 3px;
}


button#taskbar_normal image,
button#taskbar_active image,
button#taskbar_normal:hover image {
  min-width: 28px;
  min-height: 28px;
  margin-right: 3px;
}


button#taskbar_normal,
button#taskbar_active,
button#taskbar_normal:hover {
  box-shadow: none;
  -GtkWidget-hexpand: false;
  -GtkWidget-vexpand: false;
  min-height: 0px;
  border: solid 1px transparent;
  color:#e7e7e7;
}

button#taskbar_active {
  background: grey;
  border: solid 1px white;
  color: #000000;
}


button#taskbar_normal:hover,
button#taskbar_active:hover {
  background: #b4b4b4;
  border: solid 1px white;
  color: #000000;
}
```
Updated, started with 16mb is now at 40mb.

**Comment by stefonarch** [comment]
At 220mb now...

**Comment by LBCrion** [comment]
Indeed, it looks like I was a bit hasty. Appinfo had more than one memory leak. Can you please check if the current git version is behaving better?

**Comment by stefonarch** [comment]
I'm afraid not much.

**Comment by LBCrion** [comment]
Hmm, I'm running your config right now and memory footprint is completely stable. Are you sure you're on the latest git version?
Is it leaking memory while idle or is it when you're opening / closing windows or performing other operations?

**Comment by LBCrion** [comment]
Ah, I may have found it. This one would be triggered by updates to .desktop files, can be triggered by transient desktop files or containers. It should be fixed in the latest version. If you get a chance, can you please update and try again?

**Comment by stefonarch** [comment]
Looks fixed, still at 11.8mb after some time, thanks!

**Comment by LBCrion** [comment]
Cool. Thank you for all the testing on this one!
